src/Aspire.Hosting.Azure/AzureResourcePreparer.cs

```diff
@@ -466,7 +470,8 @@ ProvisioningParameter CreatePrincipalParam(string name)
             roles,
             new(() => CreatePrincipalParam(AzureBicepResource.KnownParameters.PrincipalType)),
             new(() => CreatePrincipalParam(AzureBicepResource.KnownParameters.PrincipalId)),
-            new(() => CreatePrincipalParam(AzureBicepResource.KnownParameters.PrincipalName)));
+            new(() => CreatePrincipalParam(AzureBicepResource.KnownParameters.PrincipalName)),
+            new(() => CreatePrincipalParam(AzureBicepResource.KnownParameters.ClientId)));
```

Comment:
You will need to add this to 

[https://github.com/dotnet/aspire/blob/dd294579f8be6056d789334dffd7aeffd1922790/src/Aspire.Hosting.Azure/Provisioning/Provisioners/BicepProvisioner.cs#L325-L350](https://github.com/dotnet/aspire/blob/dd294579f8be6056d789334dffd7aeffd1922790/src/Aspire.Hosting.Azure/Provisioning/Provisioners/BicepProvisioner.cs#L325-L350)